scipy/optimize/_differentialevolution.py

```diff
@@ -364,13 +364,16 @@ def __init__(self, func, bounds, args=(),
         self.__scale_arg1 = 0.5 * (self.limits[0] + self.limits[1])
         self.__scale_arg2 = np.fabs(self.limits[0] - self.limits[1])
 
-        parameter_count = np.size(self.limits, 1)
+        self.parameter_count = np.size(self.limits, 1)
+
         self.random_number_generator = _make_random_gen(seed)
 
-        #default initialization is a latin hypercube design, but there
-        #are other population initializations possible.
-        self.population = np.zeros((popsize * parameter_count,
-                                    parameter_count))
+        # default population initialization is a latin hypercube design, but
+        # there are other population initializations possible.
+        self.population = np.zeros((popsize * self.parameter_count,
+                                    self.parameter_count))
```

Comment:
Once the `DifferentialEvolutionSolver` is created, I wanted to be able to reuse the solver. This means reinitialising the population with `init_population_lhs` or `init_population_random`. This is why the methods are public, not private.  Those two methods need to know the shape of the population in order to create it.  It's not a good idea to have to pass a parameter to the methods, so I was relying on already having created a population of the right shape, which is why it's initialised there.  It is overwritten unconditionally below.

What I could do is add a `self._population_shape = (popsize * self.parameter_count, self.parameter_count)` attribute to the class.  The `init_population_xxx` method could then use that shape. However, it's 50:50 either way as to what's best.